test_fms/test_common.sh.in

```diff
@@ -49,26 +49,15 @@ oversubscribe='@OVERSUBSCRIBE@'
 run_test()
 {
     # If the tests is known to fail exit 
-    if test "x${3}" != "xskip" ; then     
-
-    # If there is no mpi launcher just ./{job_script}
-    if test "x$mpi_launcher" != "x" ; then  
-        npes="-n ${2}"
-    fi
-
-    # Check if the oversubscribe flag is turned on 
-    if test "x${4}" == "xtrue" ; then 
-    # Check if the your mpi launcher allows the oversubscribed option 
-        if "x$oversubscribe" != "x" ; then  
-            $mpi_launcher $oversubscribe $npes ./${1}
-            return 0
-        else #If you the mpi launcher doesn't allow the oversubscribed option, don't run the test
-            return 0
+    if test "x${3}" = "xskip" ; then     
+        return $TEST_SKIP
```

Comment:
Some of the shell scripts (test_mpp_domains2.sh) have multiple tests, when the script gets to any of the tests that are skipped then with this line, it is not going to continue to run the rest of the script 

you can see this in the travis ci, diag_manager and mpp_domains are marked as skip even though some of the tests did ran or were suppose to run.

Reply:
So would moving all the the skipped tests to the bottom ensure that none of the tests that are supposed to run are skipped?